### PR TITLE
Support Cypher multi-line comments

### LINKED ISSUE #1709: Cypher: inline and multi-line comments are not recognized
**Name of the lexer**

`cypher`

**Code sample**

In the following example, both `/* x */` and `/* y */` are inline comments.

```cypher
RETURN point({longitude: 2.2105491 /* x */, latitude: 48.9250016 /* y */, srid: 4326}) AS home
```

http://rouge.jneen.net/v3.26.0/cypher/UkVUVVJOIHBvaW50KHtsb25naXR1ZGU6IDIuMjEwNTQ5MSAvKiB4ICovLCBsYXRpdHVkZTogNDguOTI1MDAxNiAvKiB5ICovLCBzcmlkOiA0MzI2fSkgQVMgaG9tZQ

In the following example, the Cypher query contains a multi-line comment (and returns 42):

```cypher
RETURN /* drum
rolls */ 42
```

http://rouge.jneen.net/v3.26.0/cypher/UkVUVVJOIFwqCiovIDQyCg

### PR DESCRIPTION
**Pull request by Mogztter**
For reference, here's the Cypher EBNF related to comments:

```
Comment = ('/*', { ANY - ('*') | ('*', ANY - ('/')) }, '*/')
        | ('//', { ANY - (LF | CR) }, [CR], (LF | EOI))
        ;
```

resolves #1709

### CODE REVIEW

File: lib/rouge/lexers/cypher.rb [hunk 1]
```diff
@@ -103,6 +104,13 @@ def self.keywords
         rule %r/'(\\\\|\\'|[^'])*'/, Str::Single
         rule %r/`(\\\\|\\`|[^`])*`/, Str::Backtick
       end
+
+      state :multiline_comments do
+        rule %r(/[*]), Comment::Multiline, :multiline_comments
+        rule %r([*]/), Comment::Multiline, :pop!
+        rule %r([^/*]+), Comment::Multiline
+        rule %r([/*]), Comment::Multiline
+      end
```
Comment (Mogztter) [inline lib/rouge/lexers/cypher.rb:113]: I took inspiration from the SQL lexer: https://github.com/rouge-ruby/rouge/blob/442c8f430277889afc6ce0ccbbdbdaec687abc0f/lib/rouge/lexers/sql.rb#L132-L137

File: spec/support/lexing.rb [hunk 2]
```diff
@@ -43,5 +43,16 @@ def assert_tokens_equal(text, *expected)
       actual = lexer.lex(text).map { |token, value| [ token.qualname, value ] }
       assert { expected == actual }
     end
+
+    def assert_tokens_includes(text, *expected)
```
Comment (tancnle) [inline spec/support/lexing.rb:47]: **praise:** Thanks for adding this spec helper ❤️

File: spec/lexers/cypher_spec.rb [hunk 3]
```diff
@@ -15,4 +15,25 @@
...
+          assert_tokens_includes "RETURN point({longitude: 2.2105491 /* x */, latitude: 48.9250016 /* y */, srid: 4326}) AS home",
+                                 ["Comment.Multiline", "/* x */"],
+                                 ["Comment.Multiline", "/* y */"]
+        end
+
+        it 'handles multiline comment' do
+          assert_tokens_includes "RETURN /* drum\nrolls */ 42",
+                                 ["Comment.Multiline", "/* drum\nrolls */"],
+                                 ["Literal.Number.Integer", "42"]
+
```
Comment (tancnle) [inline spec/lexers/cypher_spec.rb:34]: **nitpick:** Could we remove this empty line?

**Review by tancnle — COMMENTED** [review]
Thank you for your contribution @Mogztter 👍🏼 The PR looks good to me. I have left some minor suggestions for your consideration. Please let me know what you think 🙏🏼

**Comment by Mogztter** [comment]
Thanks for your review, I will update the PR accordingly in the next few days 😉

**Comment by Mogztter** [comment]
@tancnle should be OK now

**Comment by tancnle** [comment]
Looks good to me. Thanks @Mogztter 🙇🏼

**Pull request merged**